packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -191,19 +191,27 @@ class TabController extends ChangeNotifier {
     _index = value;
     if (duration != null && duration > Duration.zero) {
       _indexIsChangingCount += 1;
+      _indexChangingDurations.add(duration);
+      _indexChangingCurves.add(curve!);
       notifyListeners(); // Because the value of indexIsChanging may have changed.
       _animationController!
-        .animateTo(_index.toDouble(), duration: duration, curve: curve!)
+        .animateTo(_index.toDouble(), duration: duration, curve: curve)
         .whenCompleteOrCancel(() {
           if (_animationController != null) { // don't notify if we've been disposed
-            _indexIsChangingCount -= 1;
-            notifyListeners();
-          }
-        });
+          _indexIsChangingCount -= 1;
+          _indexChangingDurations.removeLast();
+          _indexChangingCurves.removeLast();
```

Comment:
To be honest, I'm not 100% pleased with the solution but I wanted to stick close to the implementation of the indexChangingCount. Unfortunately the interaction between TabController and TabBarView doesn't seem to be optimal all in all. In the end, a more complex refactoring would be best but for my first pull request I didn't want to refactor too much :)

If you want to discuss better solutions then I'd be glad to have a call/chat.

Reply:
> In the end, a more complex refactoring would be best but for my first pull request I didn't want to refactor too much :)

Agree 100%. For instance, the refactoring I did in tabs.dart came after several PRs I files on this area and I get more confident to refactor without breaking existing tests (which is the tricky parts :smile: ).

> Unfortunately the interaction between TabController and TabBarView doesn't seem to be optimal all in all.

What makes this area looking suboptimal is that many small fixes and new features/flexibilities were introduced along the years (original PR is from 2017). How far this can be refactored is difficult to state: it might be complex because of the current expectations, but any attempt to refactor, if it does not break tests, are very welcomed!

> If you want to discuss better solutions then I'd be glad to have a call/chat.

What works the best for me is discussing based on a proposed change because it makes possible to analyze more deeply, in an asynchronous way, what is proposed (few people have a full comprehensive understanding of the inner workings, I have not yet such vision :smile: ).